fiftyone/utils/fiw.py

```
@@ -283,51 +286,331 @@ def _parse_identifier(identifier):
         scratch_dir (None): a scratch directory to use to store temporary files
         cleanup (True): whether to cleanup the scratch directory after
             extraction
+
+    Returns:
+        a tuple of (num_samples, classes)
     """
-    etau.ensure_dir(dataset_dir)
+    return parse_fiw_dataset(source_dir, dataset_dir, scratch_dir, split)
+
+
+def _validate_source_dir(source_dir):
```

Ruff (0.14.8)

Missing return type annotation for private function `_validate_source_dir`

Add return type annotation: `None`

(ANN202)